File: Basics_with_JavaScript/56 Conditional Statements/More Exersises/04. Transport Price.js
function vladi(input) {
  let distance = Number(input[0]);
  let command = String(input[1]);
  let taxiPrice = 0;
  let busPrice = 0;
  let trainPrice = 0;
  buss = false;
  train = false;
  let lowest = 0;
  let taxiPricePerKm = 0

  if (command == "day") {
    taxiPricePerKm = 0.79;
  } else if (command == "night") {
    taxiPricePerKm = 0.9;
  }

  let busPricePerKm = 0.09;
  let trainPricePerKm = 0.06;
  taxiPrice = distance * taxiPricePerKm + 0.7;
  busPrice = busPricePerKm * distance;
  trainPrice = trainPricePerKm * distance;

  if (distance < 20) {
    lowest = taxiPrice;
  } else if (distance >= 20 && distance < 100) {
    if (taxiPrice <= busPrice) {
      lowest = taxiPrice;
    } else {
      lowest = busPrice;
    }
  } else if (distance >= 100) {
    if (taxiPrice <= busPrice && taxiPrice <= trainPrice) {
      lowest = taxiPrice;
    } else if (taxiPrice <= busPrice && taxiPrice > trainPrice) {
      lowest = trainPrice;
    } else if (taxiPrice > busPrice && busPrice <= trainPrice) {
      lowest = busPrice;
    } else if (taxiPrice > busPrice && busPrice > trainPrice) {
      lowest = trainPrice;
    }
  }
  lowest = lowest.toFixed(2);

  console.log(lowest);
}

vladi(["5", "day"]);
vladi(["7", "night"]);
vladi(["25", "day"]);
vladi(["180", "night"]);

File: Basics_with_Python/PythonBasicCourse/2 Conditional Statements/6_Conditional_Statements_Excercises - Exercise/more_exercises/04. Transport Price.py
distance = int(input())
rate = input()

bus = 0
taxi = 0
train = 0

if rate == "day":
    taxi = 0.70 + 0.79 * distance
    bus = 0.09 * distance
    train = 0.06 * distance
elif rate == "night":
    taxi = 0.70 + 0.90 * distance
    bus = 0.09 * distance
    train = 0.06 * distance

if distance < 20:
    print(f"{taxi:.2f}")
elif distance >= 20 and distance < 100:
    if taxi < bus:
        print(f"{taxi:.2f}")
    else:
        print(f"{bus:.2f}")
else:
    if taxi <= bus and taxi <= train:
        print(f"{taxi:.2f}")
    elif bus <= taxi and bus <= train:
        print(f"{bus:.2f}")
    else:
        print(f"{train:.2f}")
